apollo-router-core/src/services/subgraph_service.rs

```diff
@@ -0,0 +1,62 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::Poll;
+
+use crate::prelude::graphql;
+use crate::{RouterResponse, SubgraphRequest};
+use futures::FutureExt;
+use futures::TryFutureExt;
+use http::{request, Request, Response, Uri};
+use reqwest::{Body, Url};
+use reqwest_middleware::RequestBuilder;
+use tower::{BoxError, Service};
+use tracing::info;
+use typed_builder::TypedBuilder;
+
+#[derive(TypedBuilder, Clone)]
+pub struct GraphQlSubgraphService {
```

Comment:
We already agreed on:
- respect [Rust naming guidelines](https://rust-lang.github.io/api-guidelines/naming.html): UpperCamelCase, so it's Graphql and not GraphQL nor GraphQl
- drop the prefix `Graphql` to all objects because it is already scoped in the module `apollo_core::prelude::graphql`

I'll create a ticket to make an ADR for this because I think it's not super clear for everybody.

```suggestion
pub struct GraphqlSubgraphService {
```

Reply:
Created #368